placeSpecificContent/hq/chatterboxes/chatterbox/buttons/privacyButton_server.js

```diff
@@ -103,23 +105,24 @@
 
         /* Use parenting relationships to get IDs of required entities */
         getEntityData: function() {
-            if (!button || !buttonPosition || !chatterbox) {
-                var table = Entities.getEntityProperties(_this.entityID, 'parentID').parentID;
-                Entities.getChildrenIDs(table).forEach(function(tableChildID) {
-                    var properties = Entities.getEntityProperties(tableChildID, ['position', 'name']);
-                    if (properties.name === "Chatterbox Privacy Button") {
-                        button = tableChildID;
-                        buttonPosition = properties.position;
-                        chatterbox = Entities.getEntityProperties(table, 'parentID').parentID;
-                    }
-                });
+            if (!chatterbox) {
+                var table = Entities.getEntityProperties(button, 'parentID').parentID;
+                if (table) {
+                    chatterbox = Entities.getEntityProperties(table, 'parentID').parentID;
+                }
+            }
+            if (!buttonPosition) {
+                var buttonPositionReceived = Entities.getEntityProperties(button, 'position').position;
+                if (JSON.stringify(buttonPositionReceived) !== DEFAULT_STRING_IF_ENTITY_POSITION_NOT_READ) {
```

Comment:
please check the X, Y, and Z values of what's received instead of checking against a stringified JSON object (`DEFAULT_STRING_IF_ENTITY_POSITION_NOT_READ`).